File: 3349-maximum-length-substring-with-two-occurrences/maximum-length-substring-with-two-occurrences.java
class Solution {
    public int maximumLengthSubstring(String s) {
        int[] freq = new int[26];
        int left = 0, ans = 0;

        for (int right = 0; right < s.length(); right++) {
            int i = s.charAt(right) - 'a';
            freq[i]++;

            while (freq[i] > 2) {
                freq[s.charAt(left) - 'a']--;
                left++;
            }

            ans = Math.max(ans, right - left + 1);
        }

        return ans;
    }
}

File: 3349-maximum-length-substring-with-two-occurrences/maximum-length-substring-with-two-occurrences.py
class Solution:
    def maximumLengthSubstring(self, s: str) -> int:
        freq = [0] * 26
        left = ans = 0

        for right, c in enumerate(s):
            i = ord(c) - 97
            freq[i] += 1

            while freq[i] > 2:
                freq[ord(s[left]) - 97] -= 1
                left += 1

            ans = max(ans, right - left + 1)

        return ans